pkg/azuredisk/controllerserver.go

```diff
@@ -839,6 +844,36 @@ func (d *Driver) getSnapshotByID(ctx context.Context, resourceGroup, snapshotID,
 	return generateCSISnapshot(sourceVolumeID, &snapshot)
 }
 
+// GetSourceDiskSize returns: disk size, error
+func (d *Driver) GetSourceDiskSize(resourceGroup, diskName string) (*int32, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	return d.GetSourceDiskSizeHelper(ctx, resourceGroup, diskName)
+}
+
+// GetSourceDiskSizeHelper is a helper function for GetSourceDiskSize to support recursive search for  and it returns: disk size, error
+func (d *Driver) GetSourceDiskSizeHelper(ctx context.Context, resourceGroup, diskName string) (*int32, error) {
+	result, rerr := d.cloud.DisksClient.Get(ctx, resourceGroup, diskName)
+
+	if rerr != nil {
+		return nil, rerr.Error()
+	}
+
+	if result.DiskProperties.CreationData != nil && (*result.DiskProperties.CreationData).CreateOption == "Copy" {
+		klog.V(2).Infof("Clone source disk has a parent source")
+		sourceResourceID := *result.DiskProperties.CreationData.SourceResourceID
+		parentResourceGroup, _ := GetResourceGroupFromURI(sourceResourceID)
+		parentDiskName := path.Base(sourceResourceID)
+		return d.GetSourceDiskSizeHelper(ctx, parentResourceGroup, parentDiskName)
+	}
+
+	if result.DiskProperties == nil {
```

Comment:
Shouldn't this check be placed before the if condition on line 862?

Reply:
You are right. I don't know how that ended up there haha. Thanks for pointing that out.